### PR TITLE
Fix multiple memory vulnerabilities

### PR DESCRIPTION
**Pull request by MatejKafka**
See the commit messages.

### CODE REVIEW

File: include/highlight.h [hunk 1]
```diff
@@ -189,9 +189,9 @@
 	 * @param c Symbol to be highlighted.
 	 * @param hl Highlighted Line Buffer.
 	 */
-	INLINE static int highlight_symbol(int c, char **hl)
+	INLINE static int highlight_symbol(char c, char **hl)
```
Comment (Theldus) [inline include/highlight.h:192]: If the parameter `c` is then cast to 'unsigned char', the parameter could also be 'unsigned char', right? but OK, not a problem.

**Comment by Theldus** [comment]
Hello @MatejKafka,

Thank you for delving so deep into these issues. Kat was my first 'C-parser,' and as a result, the implementation was far from ideal... After Kat, I created [Sourcery](https://github.com/Theldus/sourcery) and then [Aqua](https://gist.github.com/Theldus/09ed2205aa5ba15cdf4571b71cd1c8fc) (different tools for different purposes but sharing the same central idea, especially Aqua).

That being said, there's a lot in Kat that would benefit from a review, and maybe someday I'll rewrite the code in a way that I find more suitable.

Regarding your PR, there's just one comment I'd like to make, but it's in good shape overall, and I'll merge it. Thanks again =).

File: include/highlight.h (hunk 1 above)
Comment (MatejKafka) [reply include/highlight.h:192]: I think it's a matter of API design – if a function takes `unsigned char` and I have I `char`, I'll be wondering if it's OK to call it, or if I'm doing something wrong. The cast is an implementation detail, which imo shouldn't be reflected in the API.

Comment (Theldus) [reply include/highlight.h:192]: Yes... you're right... I should have made the buffer (`line[]`) as `unsigned char/uint8_t` in the first place...

**Pull request merged**